contrib/mesos/pkg/scheduler/podtask/roles.go

```diff
@@ -0,0 +1,61 @@
+package podtask
+
+// rolePredicate is a predicate function on role strings
+type rolePredicate func(string) bool
+
+// filterRoles filters the given slice of roles and returns a slice of roles
+// matching all given predicates
+func filterRoles(roles []string, ps ...rolePredicate) []string {
+	filtered := make([]string, 0, len(roles))
+
+next:
+	for _, r := range roles {
+		for _, p := range ps {
+			if !p(r) {
+				continue next
+			}
+		}
+
+		filtered = append(filtered, r)
+	}
+
+	return filtered
+}
+
+// duplicate returns a rolePredicate which returns true
+// if the given role is a duplicate.
+func duplicate() rolePredicate {
+	seen := map[string]struct{}{}
+
+	return func(role string) bool {
+		_, ok := seen[role]
+
+		if !ok {
+			seen[role] = struct{}{}
+		}
+
+		return ok
+	}
+}
+
+// empty returns true if the given role is empty
+func empty(role string) bool {
```

Comment:
I don't like those generic names in the podtask namespace. The functions are not about podtasks.